staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1005,6 +1146,17 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		waitGroup:           &utilwaitgroup.SafeWaitGroup{},
 	}
 
+	if r.storageVersionManager != nil {
+		// spawn storage version update in background and use channels to make handlers wait
+		processedCh := make(chan struct{})
+		ctx := apirequest.NewContext()
+		r.storageVersionManager.UpdateStorageVersion(ctx, crd, nil, processedCh)
```

Comment:
This one needs to be error aware. If we fail to write the storage version, could we fail this call and get a chance to try again?

Reply:
I removed the retry logic from [updateStorageVersionFor()](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/storageversion/updater.go#L126-L146) and moved that here for cases when UpdateStorageVersion() returns an error.

Reply:
Actually this wasn't really needed as discussed [here](https://github.com/kubernetes/kubernetes/pull/120582#discussion_r1480424305)